### ISSUE TITLE
calcite-date-picker changes height depending on months

### ISSUE THREAD

**Issue by janett-baresel** [Esri/calcite-design-system#9305]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Actual Behavior

When flipping through the months the picker changes height due to how many rows a particular month needs. This is not ideal when the button for next/previous moves up and down in certain scenarios. It is even worse if the changing height triggers different position of the popover above or below the control. (see 2 gifs).
This also can happen when changing to the next year, where the months might be shorter or longer (in number of rows) than before.

![datepicker](https://github.com/Esri/calcite-design-system/assets/11048042/d712d371-2284-40de-9149-b057d2b65ee8)

![datepicker2](https://github.com/Esri/calcite-design-system/assets/11048042/f40694a7-e0a1-44c8-82b7-07a336bb91eb)

### Expected Behavior

- [ ] The popover stays at the same place and doesn't change height so the above doesn't happen and users can seamlessly switch between months and years.
- [ ] The Date Picker always maintains space for 6 week rows. See image below.

![image](https://github.com/user-attachments/assets/36fef2bd-9c67-47d1-8a90-3cb492d91c4b)

[Calcite Design Figma file](https://www.figma.com/design/gsdKTxEvTLZTMGFNj5Mvlg/Date-Picker-changes-height-depending-on-months?node-id=0-1) available for reference. 

### Reproduction Sample

https://developers.arcgis.com/calcite-design-system/components/input-date-picker/

### Reproduction Steps

1. open input date picker sample in codepen
2. add `<div style="height:350px;"></div>` above picker
3. open picker and observe issues mentioned above.

![image](https://github.com/Esri/calcite-design-system/assets/11048042/c7868309-f9b9-4ad0-b3ab-2a44f41f0439)

### Reproduction Version

2.8.0

### Relevant Info

os and browser independent

### Priority impact

p4 - not time sensitive

### Impact

minor - just bad UX for people setting up a lot of dates for different layers. This becomes visible because SV (and MV) allow setting a visible time extent for any layer. Users will set up time ranges for multiple layers in a row and this UX makes it harder to quickly navigate to a given date.

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-angular
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Scene Viewer

**monday.com sync:** #7451041263

**Comment by geospatialem** [comment]
Design expertise sought for the following expertise 🧠 :
1. Can the component adopt a fixed height regardless of the number of week in a month?
2. If `"yes"` to `#1` _above_, should we populate more dates from the previous month?

**Comment by ashetland** [comment]
@SkyeSeitz @macandcheese This is an example from Atlassian where both of @geospatialem's questions are answered "yes". There calendar always has 6 date rows we can see that the entire bottom row here is from the next month.
![CleanShot 2024-10-22 at 15 28 16@2x](https://github.com/user-attachments/assets/f258473d-ad43-4cd7-b5f0-efba76400bf3)

**Comment by SkyeSeitz** [comment]
I agree with moving forward with a fixed height shared between months. Open to displaying the next month's dates in additional space for shorter months - a better use of space imo than leaving a potentially awkward gap. Thanks, @geospatialem and @ashetland 🙌

**Comment by ashetland** [comment]
Expected Behavior updated with final design specs.

**Comment by eriklharper** [comment]
Not sure if its possible since `input-date-picker` doesn't use `calcite-popover` internally, but the Floating UI API under the hood does allow direct control for where the popover displays.  You can force it to always display below or above if you want depending on the use case with the `placement` property https://floating-ui.com/docs/computePosition#placement

**Comment by eriklharper** [comment]
Ah, you can ignore that.  Looks like `input-date-picker` does expose a `placement` property: https://developers.arcgis.com/calcite-design-system/components/input-date-picker/#api-reference-properties-placement.  @janett-baresel have you tried setting `placement` to `bottom` to solve for that?  If so, did it work?

**Comment by DitwanP** [comment]
🍠 Verified on `3.2.0-next.53`
https://codepen.io/Ditwan-Price/pen/pvvOMby?editors=100
